central/securedclustercertgen/certificates.go

```diff
@@ -80,6 +84,13 @@ func IssueSecuredClusterCertsWithCAs(
 		if secondaryCACert.NotAfter.After(primaryCACert.NotAfter) {
 			primaryCA, secondaryCA = secondaryCA, primaryCA
 		}
+	} else {
+		// If a CA fingerprint is provided, prefer the matching CA. Otherwise just use the primary CA.
+		if sensorCAFingerprint != "" && secondaryCA != nil {
+			if sensorCAFingerprint == cryptoutils.CertFingerprint(secondaryCA.Certificate()) {
+				primaryCA, secondaryCA = secondaryCA, primaryCA
+			}
+		}
```

Comment:
nit
```suggestion
	} else if sensorCAFingerprint != "" && secondaryCA != nil && sensorCAFingerprint == cryptoutils.CertFingerprint(secondaryCA.Certificate()) {
		// If a CA fingerprint is provided, prefer the matching CA. Otherwise just use the primary CA.
		primaryCA, secondaryCA = secondaryCA, primaryCA
```